test/integration/controller/jobset_controller_test.go

```diff
@@ -170,14 +176,14 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 					// Fetch updated job objects so we always have the latest resource versions to perform mutations on.
 					// Ensure we have all expected jobs in our jobList before continuing.
 					var jobList batchv1.JobList
-					gomega.Eventually(func() (bool, error) {
-						if err := k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)); err != nil {
-							return false, err
+					gomega.Expect(k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace))).Should(gomega.Succeed())
+					if up.checkJobCreation == nil {
+						gomega.Expect(len(jobList.Items)).To(gomega.Equal(testutil.NumExpectedJobs(js)))
```

Comment:
We need to wrap this in an eventually

Reply:
we also need to use testutil.NumJobs instead of len(jobList.Items)

Reply:
updated this.